liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,256 @@
+        firmware = '{}.{}.{}.{}'.format(*tuple(msg[3:7]))
+        status.append(('Firmware version', firmware, ''))
+        return status
+
+    def set_color(self, channel, mode, colors, time_per_color=1, time_off=None,
+                  alert_threshold=_HIGH_TEMPERATURE, alert_color=[255, 0, 0],
+                  speed=3, temps=['35', '45', '55'], **kwargs):
+        """Set the color mode for a specific channel."""
+        if mode not in _PRO_COLOR_CHANGE_MODES:
+            raise KeyError('Unknown lighting mode {}'.format(mode))
+        colors = list(colors)
+        _LOGGER.debug('color count = %d', len(colors))
+        _LOGGER.debug('color change speed %d', int(speed))
+        if mode == 'alert':
+            self._write([0x5f, int(temps[0]), 0x00, int(temps[1]), 0x00, int(temps[2]), 0x00] + colors[0] + colors[1] + colors[2])
+            self._end_transaction_and_read(6)
+            self._write([0x5E, 0x01])
+            self._end_transaction_and_read(3)
+            return
+        if mode == 'fixed':
+            colors = [colors[0], colors[0]]
+        if mode != 'rainbow':
+            self._write_color(colors)
+        if mode != 'fixed':
+            self._write_color_change_speed(mode, speed)
+        self._write(_PRO_COLOR_CHANGE_MODES[mode])
+        self._end_transaction_and_read(3)
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set channel to follow a speed duty profile."""
+        if channel == 'pump':
+            raise ValueError('set_speed_profile not implemented for the pump on CorsairAsetekPro')
+        adjusted = self._prepare_profile(profile, 0, 100, _MAX_PROFILE_POINTS)
+        for temp, duty in adjusted:
+            _LOGGER.info('setting %s PWM point: (%i°C, %i%%), device interpolated',
+                        channel, temp, duty)
+        temps, duties = map(list, zip(*adjusted))
+        # Need to write curve for each fan
+        # Assume if we read n values from _get_fan_speed the fans are numbered [0, n)
+        # This will have problems on a 2 fan system when the first fan has failed
+        for i in self._get_fan_indexes(channel):
+            _LOGGER.info('setting speed for fan %d', i+1)
+            self._write([_PRO_CMD_WRITE_FAN_SPEED, i] + temps + duties)
+            self._end_transaction_and_read(32)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set channel to a fixed speed duty."""
+        if channel.startswith('fan'):
+            # Setting a fix speed is supported, however it is done by sending RPM or PWM
+            # values. I'm not 100% sure how this maps to %s used by liquidclt so set a flat
+            # speed profile instead.
+            _LOGGER.info('using a flat profile to set %s to a fixed duty', channel)
+            self.set_speed_profile(channel, [(0, duty), (_CRITICAL_TEMPERATURE - 1, duty)])
+            return
+        # Fixed speed pump is not supported, so guess a curve from the duty value given
```

Comment:
If the hardware only supports pump modes, then that's all we should support too, through `initialize --pump-mode <mode>`.

Reply:
`initialize --pump-mode <mode` already works. I'll replace this with a `NotSupportedByDevice` error